pkg/scheduler/core/spreadconstraint/select_clusters_by_cluster.go

```diff
@@ -30,9 +30,9 @@ func selectBestClustersByCluster(spreadConstraint policyv1alpha1.SpreadConstrain
 		return nil, fmt.Errorf("the number of feasible clusters is less than spreadConstraint.MinGroups")
 	}
 
-	needCnt := spreadConstraint.MaxGroups
-	if totalClusterCnt < spreadConstraint.MaxGroups {
-		needCnt = totalClusterCnt
+	needCnt := totalClusterCnt
+	if spreadConstraint.MaxGroups != 0 && spreadConstraint.MaxGroups < totalClusterCnt {
+		needCnt = spreadConstraint.MaxGroups
```

Comment:
I think `spreadConstraint.MaxGroups < totalClusterCnt` is duplicated.

Reply:
This place is where ```spreadConstraint.MaxGroups``` and ```totalClusterCnt``` are compared to obtain the smaller value.